3215 readrowsparser acceptance tests

amitchauhan1007 opened the pull request:
Updated python code file so as to be able to parse new json file

amitchauhan1007 committed:
updating .gitignore file

scotthart committed:
refactor: relocate polling_loop.h from spanner to common (#5185)

* refactor: relocate polling_loop.h from spanner to common

coryan committed:
feat(common): wrapper for bi-dir async streaming RPCs (#5183)

Implement a wrapper for bi-directional, asynchronous, streaming RPCs.

Commit:
feat(pubsub): only empty order keys for non-ordered publishers (#5187)

coryan committed:
feat(pubsub): add streaming pull function to stub (#5188)

This adds `AsyncStreamingPull()`, a new member function to start a
streaming pull request. This is a bi-directional streaming RPC. It can
(and soon will) be used to receive messages, update leases, nack
messages, and ack messages. It also implements flow control from the
server side, simplifying some of the library code.

coryan committed:
feat(pubsub): the streaming pull state machine (#5190)

Implements the state machine for streaming pulls. It initiates the
streaming pull request: creates the wrapper, then calls `Start()` and
the initial `Write()` both asynchronously. Then tracks if there is a
`Read()` and/or `Write()` call outstanding, queuing data for the next
`Write()` request if necessary. On an error, it waits for any
outstanding requests, then closes the stream gracefully, and resumes a
new streaming pull.

amitchauhan1007 committed:
Updating python code to parse new json #3215

mr-salty committed:
test: replace EXPECT_EQ(kOk) with EXPECT_STATUS_OK (#5209)

* test: replace EXPECT_EQ(kOk) with EXPECT_STATUS_OK

EXPECT_STATUS_OK prints the full status on failure instead of just the code.
In some places we were redundantly doing both and I removed the EXPECT_EQ.

mr-salty committed:
test: remove unused files (#5207)

* test: remove unused files

mr-salty committed:
test(storage): store the full Status in ThroughputResult (#5203)

This may give us more information if googleapis#5202 recurs.

since `PrintAsCsv()` now prints the full status which can contain an arbitrary message, we quote that field as specified in [rfc 4180](https://tools.ietf.org/html/rfc4180)

I also removed `throughput_result_test.cc` from `storage_benchmarks.bzl` since that's used by the `cc_library` target, and built with different
dependencies. It's already in `storage_benchmarks_unit_tests.bzl`.

mr-salty committed:
fix(spanner): use `ExecuteStreamingSql`for PDML operations (#5197)

The streaming call allows us to properly resume the operation via
`PartialResultSet::resume_token` (already handled lower in the stack).

part of #4714 (it almost fixes it, I just need to tweak some timeouts).

coryan committed:
fix(pubsub): streaming pull race conditions (#5210)

Found some race conditions when using this against production.
Apparently the timing is different enough from the emulator that these
do not show up in that case, even when running thousands of times.

Cleaned up the test a little bit too.

amitchauhan1007 committed:
removing stale entries from gitignore

mr-salty committed:
test: use `StatusIs` matcher in spanner tests

This was mostly pretty mechanical, a couple files that might deserve a bit more attention:

* bytes_test.cc: uses ContainsRegex instead of two HasSubstr calls. That imposes an order on the two substrings that did not previously exist but it LGTM and passes.
* numeric_test.cc: replaced the HasStatus matcher with the equivalent StatusIs invocations

coryan committed:
fix: MSVC+x86 build (#5212)

coryan committed:
fix: remove test from library list (#5213)

coryan committed:
ci: fix build script for Bazel (#5215)

I think I incompletely renamed a bash function. Because this feature is
rarely used this went unnoticed.

amitchauhan1007 committed:
doc: prefer UBLA references over bucket-policy-only

coryan committed:
Link directly to the quickstart guides from the top-level README. Also
include brief "make install" instructions, with some caveats, and links
to the more detailed instructions.

Commit:
doc(pubsub): Link subscriber doc to the quickstart

coryan committed:
ci: check for diffs post-build (#5219)

This can help us detect any changes introduced by the build scripts,
such as modifications to the `*.bzl` files that happen after we invoke
`ci/check-style.sh`.

coryan committed:
feat(pubsub): faster subscription shutdowns (#5198)

Finally use the `AsyncStreamingPull()` RPC to reduce the shutdown time
for subscription sessions. I have anecdotal evidence that this also
improves the overall throughput.

coryan committed:
ci: automatically fix problematic Abseil includes (#5221)

Including some Abseil headers create problems on MSVC+x86, we have
"*_quiet.h" headers to workaround those problems. This check
automatically replaces direct includes of those headers with the
"*_quiet" counterparts.

coryan committed:
cleanup(pubsub): remove AsyncPull() and friends (#5222)

Remove several member functions related to `AsyncPull()`, they were only
used in tests at this point.

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

manish-qlogic commented on google/cloud/bigtable/internal/readrowsparser_acceptance_tests.inc:
```diff
@@ -19,19 +19,19 @@
 TEST_F(AcceptanceTest, InvalidNoCommit) {
   std::vector<std::string> chunk_strings = {
```
since there are many changes in this file, we mus try to run these tests against production env.

amitchauhan1007 replied:
> since there are many changes in this file, we mus try to run these tests against production env.

Hi, the output file 'readrowsparser_acceptance_tests.inc' having extension 'inc' and there is no corresponding executable present/generated in cmake-out. Do you have any idea about the purpose of 'inc' extension and how its corresponding executable will be generated ?

amitchauhan1007 replied:
The file "readrowsparser_acceptance_tests.inc" looks like an extension of "readrowsparser_test.cc" and included in the file readrowsparser_test.cc as extended test. 
Also on running the correspoding executable "./internal_readrowsparser_test" from within cmake, all of the test are failing. 
Need to debug & fix the issue. Working on the same.

manish-qlogic commented on google/cloud/bigtable/tools/convert_acceptance_tests.py:
```diff
@@ -61,18 +61,28 @@ def camel_case(s):
-        o += '      R"chunk(\n'
-        o += "          " + s + "\n"
-        o += '        )chunk",\n'
+    if ('chunks' in t):
```
i think we previously parsing json element and adding to string, whereas new logic we are restricting to certain keys. 

it is better to use a generic approach rather then key based.

amitchauhan1007 replied:
> i think we previously parsing json element and adding to string, whereas new logic we are restricting to certain keys.
> 
> it is better to use a generic approach rather then key based.

Hi, 'chunks' in previous json file are implemented using python 'list of strings' whereas in the new json file 'chunks' are implemented as python 'dictionary' objected and hence parsed based on key-value approach. Hope it justifies the new implementation.

manish-qlogic replied:
what about converting dictionary to a list and then parsing it. That way we can accommodate any new addition in chunks?

manish-qlogic commented on google/cloud/bigtable/tools/convert_acceptance_tests.py:
```diff
@@ -82,8 +92,12 @@ def print_test(t):
-        ok = not any([r["error"] for r in t["results"]])
+    if ('results' in t):
+       	try:
+	    if ('error' in t["results"]):
```
would this exception handling create any side effect on generated code?

amitchauhan1007 replied:
> would this exception handling create any side effect on generated code?

Exception handling can be removed as its included as part of debugging code. Code works well after removing try-except also.

manish-qlogic replied:
so lets remove it,

amitchauhan1007 committed:
removing exceptional handling from code as its not required

vnghia committed:
test(storage): finish bucket and filter response (#5223)

coryan committed:
cleanup(pubsub): remove SubscriptionFlowControl stage (#5224)

The functionality left in this class did not match the name, and really
was not needed. Moved the bits that remained to
`SubscriptionMessageQueue`.

coryan committed:
cleanup(pubsub)!: simplify flow control config (#5226)

**BREAKING CHANGE**:

* Simplify the message flow control configuration. Now that the library uses
  streaming pulls, the low water marks are not used. The application developer
  simply sets limits for the number of messages (and/or bytes) outstanding.
  These limits are propagated to the service, and the service will stop
  streaming if too many messages (or bytes) are outstanding. While the Pub/Sub
  library is not GA, and this type of change is to be expected, we are close
  enough to a GA release that we think highlighting them is important.

hbchai committed:
feat(bigtable): convert protos to filters (#5232)

coryan committed:
cleanup(pubsub): simplify `SubscriptionBatchSource` (#5229)

None of the implementations requires the `size` arguments, so we can
remove it. Note that this is in the `pubsub_internal` namespace. No need
to declare this as a breaking change.

Commit:
fix(storage): Make private_key_id optional for JSON credentials (#5230)

coryan committed:
fix(pubsub): handle start completion after shutdown (#5231)

In rare circumstances the application may decide to shutdown while the
streaming RPC is starting. In that case we need to check the shutdown
manager once the streaming RPC completes, and then maybe initiate the
sequence to shutdown the streaming RPC.

mr-salty committed:
test(storage): use StatusIs in google/cloud/storage tests (#5234)

coryan committed:
test: clang-tidy fixes (#5240)

These passed the PR build because we get false positives (and therefore
ignore) `misc-unused-using-decls` in the PR.

Commit:
feat(storage): Create an example for Client::DeleteResumableUpload()

coryan committed:
feat(pubsub): an endurance benchmark (#5225)

Verify the Pub/Sub client library works well when used for hours.

coryan committed:
cleanup(pubsub)!: rename admin builders (#5235)

**BREAKING CHANGE**:

* Rename `TopicMutationBuilder` to `TopicBuilder`, and
  `SubscriptionMutationBuilder` to `SubscriptionBuilder`. This makes the
  C++ library more familiar for Cloud Pub/Sub developers coming from other
  languages.

coryan committed:
fix(pubsub): compilation problems inconsistent PRs (#5241)

I merged two PRs with no conflicts, but inconsistent code.

coryan committed:
cleanup(pubsub)!: inconsistent names in PublisherOptions (#5242)

**BREAKING CHANGE**

* Fix inconsistent naming for `PublisherOptions` attributes controlling the
  maximum number of messages per batch and the maximum number of bytes per
  batch.

coryan committed:
test(pubsub): enable DetachSubscription test (#5243)

Commit:
This fixes a few typos, makes the docs more consistent in some places and
provides up-to-date instructions for installing development tools. I removed
the manual instructions for formatting and linked to `check-style.sh` because I
think that will provide less problems and less updating in the future.  I also
reference Clang 10 as that seems to be the latest version on Fedora, which
also happens to be available in Bionic.

coryan committed:
cleanup(pubsub)!: remove redundant publisher option (#5244)

**BREAKING CHANGE**:

* Remove option to disable retries in `Publisher::Publish`. This is redundant as
  the application can set a "no retries" retry policy. This is more consistent
  with other Cloud Pub/Sub libraries. We include an example showing how to
  configure a "no retries" retry policy.

coryan committed:
cleanup(pubsub): return types for ack/nack functions (#5245)

This is a purely internal cleanup, so no ominous warnings. The functions
to ack and/or nack messages no longer return a meaningful future (they
were always returning satisfied futures). In this PR we simplify the
code by returning `void`.

coryan committed:
docs: update code of conduct to version 1.4.0 (#5250)

We are adopting a newer version of the contributor covenant.

coryan committed:
cleanup(pubsub)!: make Subscriber more consistent (#5247)

**BREAKING CHANGE**

* Change the `pubsub::Subscriber` API. A `Subscriber` is now bound to a specific
  Cloud Pub/Sub subscription, with a fixed set of `SuscriptionOptions`, just
  like a `pubsub::Publisher` is bound to a specific topic and a set of
  `PublisherOptions`. In addition to making publishers and subscribers more
  symmetrical, this makes the library more consistent with the Cloud Pub/Sub
  library for other languages. Finally, note that we are planning to rename
  `SubscriptionOptions` to `SubscriberOptions` in a future PR too.

mr-salty committed:
test: enable all clang-tidy presubmit warnings for source files (#5248)

* test: enable all clang-tidy presubmit warnings for source files

As discussed in #4230, running clang-tidy on a subset of files produces
some false positives in header files. However, the fix to that also
disabled some warnings for source files. This PR re-enables our full
set of warnings for source files, allowing presubmit to catch more issues
(see #5240 as an example of a break caused by this.)

coryan committed:
cleanup(pubsub)!: rename SubcriptionOptions (#5251)

**BREAKING CHANGE**

* Rename `pubsub::SubscriptionOptions` to `pubsub::SubscriberOptions` as these
  are bound to a specific subscriber object.

coryan committed:
refactor(pubsub): consolidate integration tests (#5252)

We had several integration tests that created a topic and subscription,
sent some messages, and then deleted the subscription and topic. Seems
better to have them all in one file, with the cleanup performed by the
`SetUp()` and `TearDown()` functions.

amitchauhan1007 committed:
updating python code file to parse new json

Commit:
cleanup: use absl::StrSplit where appropriate

coryan committed:
feat(pubsub): support subscription filters (#5253)

coryan committed:
cleanup(pubsub)!: rename SnapshotMutationBuilder (#5264)

**BREAKING CHANGE:** see the CHANGELOG.md file for details.

amitchauhan1007 committed:
updating python code to align indentation

Commit:
cleanup(pubsub): Use top-level target in Pub/Sub quickstart guide

Commit:
cleanup(storage): Update HmacKey documentation with link (#5269)

amitchauhan1007 committed:
converting code from base64 encoding

amitchauhan1007 committed:
including remaining keys required for parsing

amitchauhan1007 committed:
altering order of parsing value  key

amitchauhan1007 committed:
optimising code for parsing keys and decoding base64 string

manish-qlogic commented on google/cloud/bigtable/tools/convert_acceptance_tests.py:
```diff
@@ -24,16 +24,16 @@
 import json
 import sys
-
+import base64
```
do we have to take care of decoding in python script?

I am skeptical about it.

amitchauhan1007 committed:
refactor(bigtable): updating python code so that it can successfully parse new version of json file (#3215)

amitchauhan1007 committed:
code formatted with python black utility

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

vnghia committed:
test(storage): add rest for bucket and gcs object (#5273)

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch '3215_readrowsparser_acceptance_tests' of github.com:amitchauhan1007/google-cloud-cpp into 3215_readrowsparser_acceptance_tests
